File: Merge_Sort/Java/dalleng/MergeSort_test.java
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;
import org.junit.Assert;

import java.util.Arrays;

@RunWith(JUnit4.class)
public class MergeSort_test {

    @Test
    public void simpleSort() {
        int[] unsorted = {9, 8, 7, 6, 5, 4, 3, 2, 1};
        int[] sorted = {1, 2, 3, 4, 5, 6, 7, 8, 9};
        int[] result = MergeSort.sort(unsorted);
        String errorMsg = String.format("Failure - expected %s got %s", Arrays.toString(sorted), Arrays.toString(result));
        Assert.assertArrayEquals(errorMsg, sorted, result);
    }

}

File: Merge_Sort/Scala/timleathart/MergeSort_test.scala
object MergeSort_test {
	def main(args : Array[String]) {
	  val unsorted = List(1,5,2,6,7,4,7,3,2,2,1,2,6,8,9,9,3,6,3)
	  println(unsorted.mkString(", "))
	  
	  val sorted = MergeSort.MergeSort(unsorted)
	  println(sorted.mkString(", "))
	}
}